imjoy/imjoyPluginEngine.py

```diff
@@ -41,6 +41,13 @@ def get_drives():
         return drives
 
 
+try:
+    import pkg_resources  # part of setuptools
+
+    engine_version = pkg_resources.require("imjoy")[0].version
+except:
```

Comment:
Don't use empty `except:`. Except an error. If we know the error except an explicit error. If we can't predict the error and we know that we have to catch an error, it's ok to catch `Exception`. But that should be used as a last resort.

Reply:
OK, thanks, I will change it to catch `Exception`.
This is the case which I feel like it can fail on windows for example: https://stackoverflow.com/questions/7446187/no-module-named-pkg-resources

Reply:
Then except `ImportError`!

Reply:
Agreed, but in this case, I cannot really predict whether this other line: `pkg_resources.require("imjoy")[0].version` can also go wrong or not.
Unless you do believe this cannot go wrong, I tend to use a general Exception in this case because we don't want to crash the engine only because the version number cannot be determined (which is not critical).

Reply:
Why do we have to detect version dynamically like this? Don't we know our own version?

Reply:
Does `pkg_resources.require` only return the top distribution that was needed to fulfill the requirement, or also requirements of the requirement?

Reply:
Maybe this is not the best version to do this, but currently I have the version number placed in `setup.py`, and this is the code to read the number. Do you have better suggestion to have a single place to store the version number? maybe we can use a seprate version file?

Reply:
Out of all these options, I think I like number 4 best:
https://packaging.python.org/guides/single-sourcing-package-version/

Reply:
Will do in another PR